### PR TITLE
[CI] Update runner's label for GPU workflow

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
### Description
Using new runners require updated labels

### Checklist

- [x] CI workflows execute properly

// previe of the execution: https://github.com/oneapi-src/unified-memory-framework/actions/runs/8251864728/job/22597511622#step:1:2

### CODE REVIEW

File: .github/workflows/gpu.yml [hunk 1]
```diff
@@ -17,11 +17,10 @@ jobs:
 
     strategy:
       matrix:
-        os: ['ubuntu-22.04']
         build_type: [Release]
         compiler: [{c: gcc, cxx: g++}]
         shared_library: ['ON', 'OFF']
-    runs-on: level_zero
+    runs-on:  [ "DSS-LEVEL_ZERO", "DSS-UBUNTU" ]
```
Comment (bratpiorka) [inline .github/workflows/gpu.yml:23]: so this means that both labels are required to run this workflow, right?

Comment (lukaszstolarczuk) [reply .github/workflows/gpu.yml:23]: I hope so :wink: I'll check it yet

Comment (lukaszstolarczuk) [reply .github/workflows/gpu.yml:23]: ok, it worked perfectly: https://github.com/oneapi-src/unified-memory-framework/actions/runs/8251864728/job/22597511622#step:1:2

**Comment by lukaszstolarczuk** [comment]
As soon as we merge this PR I will remove the old runner and we start using the new one.

**Review by bratpiorka — APPROVED** [review]

**Review by PatKamin — APPROVED** [review]

**Pull request merged**